compiler/rustc_codegen_llvm/src/intrinsic.rs

```diff
@@ -437,6 +439,19 @@ impl<'ll, 'tcx> IntrinsicCallBuilderMethods<'tcx> for Builder<'_, 'll, 'tcx> {
 
                         self.call_intrinsic(llvm_name, &[llty], &[val, val, raw_shift])
                     }
+                    sym::unchecked_funnel_shl | sym::unchecked_funnel_shr => {
```

Comment:
Since this is the same implementation as `rotate_left`/`rotate_right` (except with an extra argument), it might be a bit clearer to just share the same branch for both and then add a branch that chooses what arguments to use depending on which symbol is being called.

This will make it a little clearer that they're both just funnel shifts, since that way if someone changes this code at all they don't have to change it in both places, since it's already deduped.